sklearn/ensemble/stacking.py

```diff
@@ -0,0 +1,166 @@
+"""Stacking API"""
+
+# Author: Caio Oliveira <caioaao@gmail.com>
+# License: BSD 3 clause
+
+from ..base import (BaseEstimator, TransformerMixin, MetaEstimatorMixin)
+from ..model_selection import cross_val_predict
+from ..pipeline import (make_union, make_pipeline)
+from ..preprocessing import FunctionTransformer
+
+
+class BlendedEstimator(BaseEstimator, MetaEstimatorMixin, TransformerMixin):
+    """Transformer to turn estimators into blended estimators
+
+    This is used for stacking models. Blending an estimator prevents data leaks
+    between the model layers. Blending will happen only when calling
+    `fit_transform`, as it's the only stage where this makes sense.
+
+    Parameters
+    ----------
+    base_estimator: the estimator to be blended.
+
+    cv: cv to be used, optional, default: 3
+        Will be passed to `cross_val_predict` during `fit_transform`.
+
+    method: string, optional, default: 'auto'
+        Invokes the passed method name of the passed estimator. If the method
+        is `auto`, will try to invoke `predict_proba` or `predict` in that
+        order.
+
+    n_jobs: int, optional, default: 1
+        Number of jobs to be passed to `cross_val_predict` during
+        `fit_transform`.
+    """
+    def __init__(self, base_estimator, cv=3, method='auto', n_jobs=1):
+        self.base_estimator = base_estimator
+        self.cv = cv
+        self.method = method
+        self.n_jobs = n_jobs
+
+    def fit(self, *args, **kwargs):
+        self.base_estimator = self.base_estimator.fit(*args, **kwargs)
```

Comment:
I think we should raise `NotImplementedError` if this is not going to behave the same as `fit_transform`.